tools/benchmark/src/main/java/com/datadog/benchmark/internal/reader/MemoryVitalReader.kt

```diff
@@ -7,16 +7,41 @@
 package com.datadog.benchmark.internal.reader
 
 import java.io.File
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
 
 /**
  * Reads the device's `VmRSS` based on the `/proc/self/status` file.
  * cf. documentation https://man7.org/linux/man-pages/man5/procfs.5.html
  */
 internal class MemoryVitalReader(
-    internal val statusFile: File = STATUS_FILE
+    internal val statusFile: File = STATUS_FILE,
+    private val readerDelay: Long = STATUS_FILE_READ_INTERVAL_MS
 ) : VitalReader {
 
-    override fun readVitalData(): Double? {
+    private var maxMemory: Double = 0.0
+    private val executorService: ScheduledExecutorService = Executors.newSingleThreadScheduledExecutor()
+
+    private val readRunnable: Runnable = Runnable {
+        maxMemory = (readStatusFile() ?: 0.0).coerceAtLeast(maxMemory)
```

Comment:
it is a bit weird the way we use `coerceAtLeast` here as there should be `minMemory` instead of `max`

Reply:
actually this is the function proposed by Kotlin compiler to convert if I type `Math.max`

Reply:
yes but I would expect to be `coerceAtLeast(minMemory)`

Reply:
`coerceAtLeast` ensures the this value is not less than the current `maxMemory`, if I reverse the caller and param, will it looks better to you?
```
 maxMemory = maxMemory.coerceAtLeast(readStatusFile()?:0.0)
```

Reply:
I agree with @mariusc83 , the doc of `coerceAtLeast` states 
> Ensures that `this` value is not less than the specified minimumValue.
> Returns: `this` value if it's greater than or equal to the minimumValue or the minimumValue otherwise.
So by definition you should call it with the `minMemory`

Reply:
Yes, I understand, but here the min value should be the current `maxMemory`, it ensures the next `maxMerory` is not less than the current one. I found it's a bit weird to rename it to `minMemory` because it doesn't stand for what it actually means. Do you have any suggestion on that, or maybe I just use the `Math.max` to avoid the confusion?
```
maxMemory = Math.max(readStatusFile()?:0.0, maxMemory)
```

Reply:
Understood, indeed it makes more sense to use Max rather than coerceAtLeast. 

Why don't you use the static `kotlin.math.max` fun then instead of the Java `Math.max`?

Reply:
you are right, static kotlin max is more understandable and it doesn't throw compiler warning, updated!